src/vs/workbench/contrib/preferences/browser/preferencesActions.ts

```diff
@@ -181,8 +184,9 @@ export class OpenWorkspaceSettingsAction extends Action {
 		this.enabled = this.workspaceContextService.getWorkbenchState() !== WorkbenchState.EMPTY;
 	}
 
-	run(event?: any): Promise<any> {
-		return this.preferencesService.openWorkspaceSettings();
+	run(args: any): Promise<any> {
+		const query = typeof args === 'string' ? args : undefined;
+		return this.preferencesService.openWorkspaceSettings(query ? false : undefined, { query });
```

Comment:
I think we already support this for settings.

https://github.com/microsoft/vscode/blob/6eaf748366a48a5b6f9645cfa5e2257f51a027a6/src/vs/workbench/contrib/preferences/browser/preferences.contribution.ts#L230-L240

So not sure if you want to do this for all commands. @roblourens FYI